src/libmongoc/src/mongoc/mongoc-topology-background-monitoring.c

```diff
@@ -311,10 +311,12 @@ _mongoc_topology_background_monitoring_stop (mongoc_topology_t *topology)
    }
 
    /* Signal all RTT monitors to shut down. */
+   bson_mutex_lock (&topology->tpld_modification_mtx);
    for (size_t i = 0u; i < n_rtt_monitors; i++) {
       server_monitor = mongoc_set_get_item (topology->rtt_monitors, i);
       mongoc_server_monitor_request_shutdown (server_monitor);
    }
+   bson_mutex_unlock (&topology->tpld_modification_mtx);
```

Comment:
I'm not sure if this mutex is relevant here. Do you have a link to the Coverity warning related to these lines?

Reply:
I sent you the link in a DM.